test/any/Diff-test.js

```diff
@@ -0,0 +1,961 @@
+      // Expecting left to be undefined and right to be defined.
+      expect(chunks[0].leftValue).toBeUndefined();
+      expect(chunks[0].rightValue).toBeDefined();
+    });
+
+    it('should return a diff fragment if one has a member and one does not', function() {
+      var leftSrc = 'TestSrc1';
+      var rightSrc = 'TestSrc2';
+      var interfaceName = 'Test';
+      var firstMap = createMap(`interface ${interfaceName} { serializer; };`, leftSrc);
+      var secondMap = createMap(`interface ${interfaceName} {};`, rightSrc);
+
+      var chunks = differ.diff(firstMap, secondMap);
+      expect(chunks.length).toBe(1);
+      // Expecting definitionName field to be name of interface.
+      expect(chunks[0].definitionName).toBe(interfaceName);
+      // Expecting status to represent no match.
+      expect(chunks[0].status).toBe(DiffStatus.NO_MATCH_ON_RIGHT);
+      // Expecting difference to be in members field of definition.
+      expect(chunks[0].propPath).toBe('.definition.members');
+      // Expecting left to be defined, and right undefined.
+      expect(chunks[0].leftValue).toBeDefined();
+      expect(chunks[0].rightValue).toBeUndefined();
+      // Expecting left source and right source to be correctly populated.
+      expect(chunks[0].leftSources).toBeDefined();
+      expect(chunks[0].leftSources.length).toBe(1);
+      expect(chunks[0].leftSources[0]).toBe(leftSrc);
+      expect(chunks[0].rightSources).toBeDefined();
+      expect(chunks[0].rightSources.length).toBe(1);
+      expect(chunks[0].rightSources[0]).toBe(rightSrc);
+    });
+
+    it('should return a diff fragment if members have different values', function() {
+      var firstMap = createMap(`interface Test { const boolean isPotato = true; };`);
+      var secondMap = createMap(`interface Test { const boolean isPotato = false; };`);
+
+      var chunks = differ.diff(firstMap, secondMap);
+      expect(chunks.length).toBe(1);
+      // Expecting difference to be in the value field of the member isPotato.
+      expect(chunks[0].propPath).toBe('.definition.members[isPotato].member.value.literal');
+      expect(chunks[0].status).toBe(DiffStatus.VALUES_DIFFER);
+      expect(chunks[0].leftValue).toBe('true');
+      expect(chunks[0].rightValue).toBe('false');
+    });
+
+    it('should return a diff fragment if members have different types', function() {
+      var firstMap = createMap(`interface Test { attribute boolean isPotato; };`);
+      var secondMap = createMap(`interface Test { attribute unsigned long isPotato; };`);
+
+      var chunks = differ.diff(firstMap, secondMap);
+      expect(chunks.length).toBe(1);
+      // Expecting difference to be in the type field of member.
+      expect(chunks[0].propPath).toBe('.definition.members[isPotato].member.type.name.literal');
+      expect(chunks[0].status).toBe(DiffStatus.VALUES_DIFFER);
+      expect(chunks[0].leftValue).toBe('boolean');
+      expect(chunks[0].rightValue).toBe('unsigned long');
```

Comment:
This has actually been fixed by another change I made. This was what was happening previously:

Let built in types be defined as `('unsigned', 'short', 'long', 'unrestricted', 'float', 'double', 'byte', 'octet)+` and custom type be all other types.

When comparing two definitions with built in types (e.g. `attribute boolean isPotato;` vs `attribute unsigned long isPotato;`), the chunk will be defined as follows:
```
propPath: '.definition.members['isPotato'].member.type.name.literal'
leftValue: 'boolean'
rightValue: 'unsigned long`
```


When comparing two definitions with custom types (e.g. `attribute Point unit;` vs  `attribute Vector unit;`), the chunk will be defined as follows:
```
propPath: '.definition.members['unit'].members.type.name.literal',
leftValue: 'Point',
rightValue: 'Vector'
```


However, when comparing definitions of built in type vs custom type (e.g. `attribute long potato` vs `attribute Vector potato`), the chunk will be defined as follows:
```
propPath: '.definition.members['unit'].members.type',
leftValue: <An AST node of type Literal>,
rightValue: <An AST node of type Identifier.
```

However, the diff algorithm has been altered to diff as far as possible, assuming the properties of the AST nodes are the same. Thus, it should now be fine.